### ISSUE TITLE
Add a utility class for white-space: nowrap

### ISSUE THREAD

**Issue by mikewheaton** [OfficeDev/Office-UI-Fabric#365]
Paul Schaeflein's [Office UI Fabric - first impressions
](http://www.schaeflein.net/office-ui-fabric-first-impressions/) article mentions the need for a `white-space: nowrap;` utility. We could create `.ms-u-noWrap` to help with this. We may also want to have one that adds truncation with an ellipsis, as we use this in many places.

**Comment by Jahnp** [comment]
Both of these make sense and are pretty low-hanging fruit.

**Comment by StfBauer** [comment]
Wouldn't it be better to have ```table-layout: fixed``` in this case ?

**Comment by mikewheaton** [comment]
I think `table-layout: fixed` applies only to `table` elements, where it forces the columns to respect the set width rather than stretching to accommodate the content. See [this example](http://www.w3schools.com/cssref/tryit.asp?filename=trycss_table_table-layout). So within a table it would stop text from wrapping or overflowing, but I don't think it works inside of a typical `div`. Or am I missing something?

**Comment by StfBauer** [comment]
No you are right. When you use table-layout: fixed the table renders on first row and not when the table have been fully loaded.
What I ment was I you have table columns that shouldn't wrap it might make more sense to have it on a higher level rather than specify it on columns individually.

```
table.fixed{
    table-layout: fixed;

   td{
       white-space: nowrap;
   }
}
```
Just maybe.

**Comment by pschaeflein** [comment]
In my scenario, I did not want the column width fixed, rather I wanted the column width to expand so that the text did not wrap. Seems that Stf's comment is the opposite. (Although I'm not a CSS expert, so I may be understanding it wrong.)

**Comment by mikewheaton** [comment]
If we're trying to modify the behavior specific to tables, I think we'd want to introduce a modifier like `.ms-Table--fixedWidth` to specifically apply the `table-layout: fixed;` property. We'd have to look at common scenarios and make sure we have examples explaining this clearly.

For text outside of tables, where we want to prevent wrapping to another line and show ellipses (...) instead, a more general `.ms-u-noWrap` utility could be created.

This way we're clear that one is a table-specific behavior, and the other a general utility that can be used almost anywhere.

**Comment by mikewheaton** [comment]
We discussed this today and decided to create a new utility, `.ms-u-textTruncate` which can be used to prevent text from wrapping to a second line and show an ellipsis (...) to indicate that the text has been truncated. This issue will now track adding that utility class.

As for table layouts, we now have a [separate issue](https://github.com/OfficeDev/Office-UI-Fabric/issues/386) to track the work of adding a `.ms-Table--fixed` modifier.

**Comment by pschaeflein** [comment]
That all sounds fine, but what if I want the table column to expand so that the text doesn't wrap and I don't want ellipses?

- Paul

Thumbed from my phone. Please forgive brevity and typos.



On Wed, Feb 24, 2016 at 2:36 PM -0800, "Mike Wheaton" <notifications@github.com<mailto:notifications@github.com>> wrote:


We discussed this today and decided to create a new utility, .ms-u-textTruncate which can be used to prevent text from wrapping to a second line and show an ellipsis (...) to indicate that the text has been truncated. This issue will now track adding that utility class.

As for table layouts, we now have a separate issue<https://github.com/OfficeDev/Office-UI-Fabric/issues/386> to track the work of adding a .ms-Table--fixed modifier.

-

**Comment by mikewheaton** [comment]
The default table behavior is for the column to expand as wide as it can (without causing another column to wrap) and then for the text to wrap.

You want the columns to auto-size (`table-layout: auto;`) but within a cell, you want to limit it to one line without the ellipses? For that you should be able to use `white-space: nowrap;` to keep it to a single line, and `overflow:hidden;' so that the excess text is not visible. But how will users know that they're not seeing all of the content?

**Comment by mikewheaton** [comment]
Fixed by #534.
